clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/job/KubectlJobExecutor.java

```diff
@@ -577,6 +578,59 @@ private String getOAuthToken(KubernetesV2Credentials credentials) {
     return status.getOutput();
   }
 
+  public Set<KubernetesKind.ScopedKind> apiResources(KubernetesV2Credentials credentials) {
+    List<String> command = kubectlAuthPrefix(credentials);
+    command.add("api-resources");
+
+    JobResult<String> status = jobExecutor.runJob(new JobRequest(command));
+
+    // api-resources can return a non-zero status code but still return data
+    // log here as a warning
+    if (!status.getResult().equals(JobResult.Result.SUCCESS)) {
+      log.warn("There was an error reading api-resources. All available kinds may not be present.");
+    }
+
+    String output = status.getOutput().trim();
+    if (StringUtils.isEmpty(output)) {
+      return new HashSet<>();
+    }
+
+    return KubernetesApiResourceParser.parse(output);
+  }
+
+  public boolean authCanI(KubernetesV2Credentials credentials, String kind, String verb) {
+    List<String> command = kubectlAuthPrefix(credentials);
+    command.add("auth");
+    command.add("can-i");
+    command.add(verb);
+    command.add(kind);
+
+    JobResult<String> status = jobExecutor.runJob(new JobRequest(command));
+
+    if (!status.getResult().equals(JobResult.Result.SUCCESS)) {
```

Comment:
Minor: This could just be return `status.getResult().equals(JobResult.Result.SUCCESS)` to avoid the `if` (and same for the next function).  Also, for enums where we can't ever create new ones on the fly, it's generally safe to compare with `==` (unlike some places where we've overused this for classes that can be created on the fly).